src/steps/owners/index.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  createDirectRelationship,
+  IntegrationStep,
+  IntegrationStepExecutionContext,
+  RelationshipClass,
+} from '@jupiterone/integration-sdk-core';
+import { createAPIClient } from '../../client';
+import { IntegrationConfig } from '../../config';
+import { getEntityKey } from '../../utils';
+import { Entities, IntegrationSteps, Relationships } from '../constants';
+import { createUserEntity } from './converters';
+
+export async function fetchOwners({
+  instance,
+  executionHistory,
+  jobState,
+}: IntegrationStepExecutionContext<IntegrationConfig>) {
+  const apiClient = createAPIClient(instance.config, executionHistory);
+  await apiClient.iterateOwners(async (owner) => {
+    const { roleId } = await apiClient.fetchUser(owner.userId.toString());
+
+    const roleEntity = await jobState.findEntity(
+      getEntityKey(Entities.ROLE, roleId),
+    );
+    const userEntity = createUserEntity(owner);
+    await jobState.addEntity(userEntity);
```

Comment:
Can we split this into two separate steps? One for fetching owners and adding the users to the job state, the other for fetching user details (for the role) and building the relationship between the role entity? If we fail to fetch user details, we won't ingest any owners at all.

Does the `fetchUser` API require a different OAuth scope than fetching owners? That would be another reason to split these up.

Reply:
Another advantage is that this will result in more focused tests. The `fetchOwners` test will only consider owner entities instead of spanning relationships.